x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/PolicyStepsRegistry.java

```diff
@@ -69,6 +70,7 @@ public void update(ClusterState currentState, Client client, LongSupplier nowSup
                     stepMap.put(policy.getName(), new HashMap<>());
                     Map<Step.StepKey, Step> stepMapForPolicy = stepMap.get(policy.getName());
                     for (Step step : policyAsSteps) {
+                        assert ErrorStep.NAME.equals(step.getKey().getName()) == false;
```

Comment:
yeah I thought about throwing an exception here too but settled on an assert at least for now because if we throw an exception we will stop the update for all policies not just the one with the error. As you say, we control all the steps at the moment so I think this is enough for now